optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/bi/DroolsAbstractBiConstraintStream.java

```diff
@@ -190,6 +182,8 @@ public final Constraint impactScoreConfigurableBigDecimal(String constraintPacka
         }
     }
 
+    protected abstract DroolsAbstractConstraintStream<Solution_> getParent();
```

Comment:
I would expect the parent of a DroolsBi to return a DroolsBi instead of a general CS. Can the return type be more specific? I haven't checked in depth, could be this is wrong.
Sometimes it could be null (for example the join CS with 2 bridge parents maybe?)

Reply:
Sometimes the parent of a `Bi` is a `Uni`. (See `Uni.groupBy(groupKeyMapping, collector)`.)